sawmill-core/src/test/java/io/logz/sawmill/utils/FactoryUtils.java

```diff
@@ -14,13 +15,16 @@
 import java.util.Map;
 
 public class FactoryUtils {
+
     public static final TemplateService templateService = new TemplateService();
-    public static final ConditionParser conditionParser;
+    public static final ProcessorFactoryRegistry defaultProcessorFactoryRegistry = new ProcessorFactoryRegistry(
+            new ProcessorFactoriesLoader(templateService)
+    );
+    public static final ConditionFactoryRegistry defaultConditionFactoryRegistry = new ConditionFactoryRegistry(
+            new ConditionalFactoriesLoader(templateService)
+    );
+    public static final ConditionParser conditionParser = new ConditionParser(defaultConditionFactoryRegistry);
```

Comment:
So we still have singleton instances, just in a different class? This code is very confusing. We had several different static instances in a few places, and now we moved them here?

Since we had static variables here before, we might as well keep this new code.
But it would have been better if all of these singletons were fields in a single SawmillContext object. That way you could instantiate your own instance for a test or for real usage and pass this and not require using static variables. at all.

Reply:
@barakm, this is _**TEST**_ sources....

And thats the changes @naorgu suggested....

Reply:
bah, I mis-read. My mistake.